index.js

```diff
@@ -46,6 +48,22 @@ module.exports = class TrailsApp extends events.EventEmitter {
     this.setMaxListeners(app.config.main.maxListeners || 128)
   }
 
+  buildLog(config) {
+    const logger = config.log.logger
+    if (logger.levels) {
```

Comment:
I don't mind the general concept of this feature, but this logic is very winston-specific. How many other loggers have a `.levels` property on the log object? I'm going to keep this PR open but let's see if we can think of a more generic way to achieve this.